src/quo2/components/wallet/token_input/view.cljs

```diff
@@ -0,0 +1,67 @@
+(ns quo2.components.wallet.token-input.view
+  (:require
+    [clojure.string :as string]
+    [quo2.components.buttons.button.view :as button]
+    [quo2.components.markdown.text :as text]
+    [quo2.foundations.colors :as colors]
+    [quo2.foundations.resources :as resources]
+    [quo2.theme :as quo.theme]
+    [react-native.core :as rn]
+    [reagent.core :as reagent]
+    [quo2.foundations.common :as common]
+    [quo2.components.wallet.token-input.style :as style]))
+
+(defn calc-value
+  [crypto? currency token value conversion]
+  (if crypto?
+    (str (get common/currency-label currency) (.toFixed (* value conversion) 2))
+    (str (.toFixed (/ value conversion) 2) " " (string/upper-case (clj->js token)))))
+
+(defn- view-internal
+  []
+  (let [width     (:width (rn/get-window))
+        value     (reagent/atom 0)
+        crypto?   (reagent/atom true)
+        input-ref (atom nil)]
+    (fn [{:keys [theme token currency conversion]}]
+      [rn/view {:style (style/main-container width)}
+       [rn/view {:style style/amount-container}
+        [rn/pressable
+         {:on-press #(.focus ^js @input-ref)
```

Comment:
I'm sorry @OmarBasem. I noticed it now.

Can you update the `on-press` function to check for the input ref and focus? Calling the `focus` without a check is error-prone.

```suggestion
         {:on-press (fn [] 
                       (some-> ^js @input-ref
                               .focus))
```